internal/uvm/vsmb.go

```diff
@@ -107,6 +108,16 @@ func (uvm *UtilityVM) AddVSMB(ctx context.Context, hostPath string, guestRequest
 			},
 			ResourcePath: vSmbShareResourcePath,
 		}
+
+		if guestRequest != "" {
+			if gr, ok := guestRequest.(guestrequest.GuestRequest); ok {
+				if settings, ok := gr.Settings.(*hcsschema.MappedDirectory); ok {
+					settings.HostPath = share.name
+				}
+			}
+			modification.GuestRequest = guestRequest
+		}
+
```

Comment:
Github won't let me add a comment any lower, but should we be returning an error when 
`if requestType == requesttype.Add || options.RestrictFileAccess {` is not true?

Reply:
The scenario where this condition isn't met seems to be when the path being passed in is a directory and the share is already mounted so we're just upping the ref count so I don't believe so.

Reply:
This has nothing to do with the guest request correct, you're referring to AddVSMB in general?

Reply:
Correct.

Reply:
Ok cool, then no I don't think we need to return an error